### PR TITLE
Fix-#363 'type should be enum'

### LINKED ISSUE #363: Make the type of a Task a enum of values
In `/Sources/MeiliSearch/Model/Task.swift`:

The attribute `public let type: String` should become an enum.
The possible values are: 
```
indexCreation, indexUpdate, indexDeletion, indexSwap, documentAdditionOrUpdate, documentDeletion, settingsUpdate, dumpCreation, taskCancelation, taskDeletion, and snapshotCreation
```

### PR DESCRIPTION
**Pull request by Dishant10**
# Pull Request

## Related issue
Fixes #363 

## What does this PR do?
Makes the type variable an enum and all the cases are made as per the issue description.
- ...

## PR checklist
Please check if your PR fulfills the following requirements:
- [• ] Does this PR fix an existing issue, or have you listed the changes applied in the PR description (and why they are needed)?
- [• ] Have you read the contributing guidelines?
- [• ] Have you made sure that the title is accurate and descriptive of the changes?

Thank you so much for contributing to Meilisearch!

### CODE REVIEW

**Review by brunoocasali — CHANGES_REQUESTED** [review]
Hi @Dishant10, thanks for your PR!

The CI must pass in order to merge your contribution! Check the linter messages and make sure your tests are passing locally :)

**Comment by Dishant10** [comment]
@brunoocasali can you please guide me what do I have to do in order to fix the other two failing tests.
Thank you!!

File: Sources/MeiliSearch/Model/Task.swift [hunk 1]
```diff
@@ -21,7 +21,20 @@ public struct Task: Codable, Equatable {
   public let status: Task.Status
 
   /// Type of the task.
-  public let type: String
+  //public let type: String
+    public enum Type:Codable {
+        case indexCreation
```
Comment (brunoocasali) [inline Sources/MeiliSearch/Model/Task.swift:26]: The indentation is not following the guidelines of the project. Also, you can move this enum to the bottom of the file but still inside of the scope of the `public struct Task`.

`Type` is a protected word in Swift, so you can use `TaskType` instead.

File: Sources/MeiliSearch/Model/Task.swift [hunk 2]
```diff
@@ -21,7 +21,20 @@ public struct Task: Codable, Equatable {
...
+        case indexUpdate
+        case indexDeletion
+        case indexSwap
+        case documentAdditionOrUpdate
+        case documentDeletion
+        case settingsUpdate
+        case dumpCreation
+        case taskCancelation
+        case taskDeletion
+        case snapshotCreation
```
Comment (brunoocasali) [inline Sources/MeiliSearch/Model/Task.swift:36]: You still have to fix these typing errors to make your tests pass:

image

Example, from this:  

`XCTAssertEqual("indexCreation", task.type)` 
to this:
`XCTAssertEqual(Task.TaskType.indexCreation, task.type)`

File: Sources/MeiliSearch/Model/Task.swift [hunk 3]
```diff
@@ -21,7 +21,20 @@ public struct Task: Codable, Equatable {
   public let status: Task.Status
 
   /// Type of the task.
-  public let type: String
+  //public let type: String
+    public enum Type:Codable {
```
Comment (brunoocasali) [inline Sources/MeiliSearch/Model/Task.swift:25]: Also, there is some GithubAction annotations, please fix them:

image

**Review by brunoocasali — CHANGES_REQUESTED** [review]
I left some comments they may help you fix the issues!

**Comment by Dishant10** [comment]
Thank you! @brunoocasali this will help me a lot!
I will fix these as soon as possible!
Thank you!

**Comment by Dishant10** [comment]
I also had some doubts while fixing typing issues as you mentioned in the IndexesResults.swift file.  So I want to test it locally first.

**Comment by brunoocasali** [comment]
> Hii @brunoocasali I'm sorry I was away for some time. I've made some changes, how can I test and get the messages on my local before raising a PR. ? Thank you!!

Hi @Dishant10 good to see you here!

You can follow the instructions written here:
https://github.com/meilisearch/meilisearch-swift/blob/main/CONTRIBUTING.md#development-workflow

**Comment by brunoocasali** [comment]
Closing due to inactivity!

**Pull request closed without merging**